pkg/monitor/monitorapi/identification_pod.go

```diff
@@ -73,57 +73,168 @@ func (r ContainerReference) ToLocator() string {
-	annotations := map[string]string{}
+	annotations := map[AnnotationKey]string{}
 	for _, curr := range tokens {
 		if !strings.Contains(curr, "/") {
-			continue
+			return annotations
 		}
 		annotationTokens := strings.Split(curr, "/")
-		annotations[annotationTokens[0]] = annotationTokens[1]
+		annotations[AnnotationKey(annotationTokens[0])] = annotationTokens[1]
 	}
 	return annotations
 }
 
-func ReasonFrom(message string) string {
+func NonAnnotationMessage(message string) string {
+	tokens := strings.Split(message, " ")
+	for i, curr := range tokens {
+		if !strings.Contains(curr, "/") {
+			return strings.Join(tokens[i:], " ")
+		}
+	}
+	return ""
+}
+
+func ReasonFrom(message string) EventReason {
 	annotations := AnnotationsFromMessage(message)
-	return annotations["reason"]
+	return EventReason(annotations[AnnotationReason])
 }
 
 func PhaseFrom(message string) string {
 	annotations := AnnotationsFromMessage(message)
-	return annotations["phase"]
+	return annotations[AnnotationPodPhase]
+}
+
+type EventReason string
+
+const (
+	IPTablesNotPermitted EventReason = "iptables-operation-not-permitted"
+
+	DisruptionBeganEventReason              EventReason = "DisruptionBegan"
+	DisruptionEndedEventReason              EventReason = "DisruptionEnded"
+	DisruptionSamplerOutageBeganEventReason EventReason = "DisruptionSamplerOutageBegan"
+
+	HttpClientConnectionLost EventReason = "HttpClientConnectionLost"
+
+	PodReasonCreated               EventReason = "Created"
+	PodReasonGracefulDeleteStarted EventReason = "GracefulDelete"
+	PodReasonForceDelete           EventReason = "ForceDelete"
+	PodReasonDeleted               EventReason = "Deleted"
+	PodReasonScheduled             EventReason = "Scheduled"
+
+	ContainerReasonContainerExit      EventReason = "ContainerExit"
+	ContainerReasonContainerStart     EventReason = "ContainerStart"
+	ContainerReasonContainerWait      EventReason = "ContainerWait"
+	ContainerReasonReadinessFailed    EventReason = "ReadinessFailed"
+	ContainerReasonReadinessErrored   EventReason = "ReadinessErrored"
+	ContainerReasonStartupProbeFailed EventReason = "StartupProbeFailed"
+	ContainerReasonReady              EventReason = "Ready"
+	ContainerReasonNotReady           EventReason = "NotReady"
+
+	PodReasonDeletedBeforeScheduling EventReason = "DeletedBeforeScheduling"
+	PodReasonDeletedAfterCompletion  EventReason = "DeletedAfterCompletion"
+)
+
+type AnnotationKey string
+
+const (
+	AnnotationReason            AnnotationKey = "reason"
+	AnnotationContainerExitCode AnnotationKey = "code"
+	AnnotationCause             AnnotationKey = "cause"
+	AnnotationNode              AnnotationKey = "node"
+	AnnotationConstructed       AnnotationKey = "constructed"
+	AnnotationPodPhase          AnnotationKey = "phase"
+	AnnotationIsStaticPod       AnnotationKey = "mirrored"
+	// TODO this looks wrong. seems like it ought to be set in the to/from
+	AnnotationDuration AnnotationKey = "duration"
```

Comment:
We're indicating here this fixed set of things we can annotate in a message, packing these into a single string then can get messy to pull out on the other side as soon as someone puts in a space. (i.e. e2e-test) Could we model this with actual json for the fixed set of things we support, omit if empty? I think this would carry benefits in the charting tools, and when we are pushing these into loki (very soon).

I imagined that we'd do similar for locator.

i.e. 
```json
       {
            "level": "Info",
            "locator": "ns/openshift-cluster-csi-drivers pod/gcp-pd-csi-driver-node-89rm9 uid/bdb45aa3-4807-4f1d-afe6-5c44cbc35bb3 container/csi-liveness-probe",
            "message": "constructed/true reason/ContainerWait missed real \"ContainerWait\"",
            "from": "2023-05-25T21:09:34Z",
            "to": "2023-05-25T21:37:37Z"
        },
```

would become:

```json
       {
            "level": "Info",
            "locator": {
                   "ns": "openshift-cluster-csi-drivers pod/gcp-pd-csi-driver-node-89rm9",
                   "uid": "bdb45aa3-4807-4f1d-afe6-5c44cbc35bb3",
                   "container": "csi-liveness-probe",
            },
            annotations: {
                   "constructed": "true",
                   "reason": "ContainerWait",
            },
            "message": "missed real \"ContainerWait\"",
            "from": "2023-05-25T21:09:34Z",
            "to": "2023-05-25T21:37:37Z"
        },
```

Reply:
@dgoodwin this is a step towards that.  
1. identification
2. structure
3. swap: https://github.com/openshift/origin/pull/27947

Reply:
I follow now, good idea.